### PR TITLE
[CPU] Interpolate runtime params cache

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *interpolate runtime params cache*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/mkldnn_eltwise_node.cpp [hunk 1]
```diff
@@ -1934,6 +1934,7 @@ void MKLDNNEltwiseNode::prepareParams() {
     // together with the corresponding appendPostOps method to pass the scales and shifts pointers at runtime.
     // Until then we have to read them from the quantization_t directly, store them somewhere
     // and nullify them to get read of the address dependency in the key structure
+    fqDataPtrs.clear();
```
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/mkldnn_eltwise_node.cpp:1937]: @maxnick , I think we need reset fqDataPtrs first here to remove fqDataPtrs of previously stored, as it's always passed from beginning element[(args_ptrs.post_op_data = fqDataPtrs.data())](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/mkldnn_eltwise_node.cpp#L1999) for current execution. Do you agree?

**Comment by chenhu-wang** [comment]
@maxnick @dmitry-gorokhov , could you please help review?

File: src/plugins/intel_cpu/src/nodes/mkldnn_eltwise_node.cpp (hunk 1 above)
Comment (maxnick) [reply src/plugins/intel_cpu/src/nodes/mkldnn_eltwise_node.cpp:1937]: Yes, of course!!! Nice catch!

**Comment by dmitry-gorokhov** [comment]
@maxnick please review the PR

**Review by maxnick — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp [hunk 2]
```diff
@@ -2038,21 +2130,73 @@ void MKLDNNInterpolateNode::prepareParams() {
...
+    auto result = cache->getOrCreate(key, buildExecutor);
+    execPtr = result.first;
+
     lastOutputDims = dstDims;
+
+    // if (result.second == CacheEntryBase::LookUpStatus::Hit) {
+    //     std::cout << "Interpolate: LookUpStatus::Hit" << std::endl;
+    // } else {
+    //     std::cout << "Interpolate: LookUpStatus::Miss" << std::endl;
+    // }
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp]: done, thanks Maksim.

File: src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp [hunk 3]
```diff
@@ -2038,21 +2130,73 @@ void MKLDNNInterpolateNode::prepareParams() {
...
+            auto &weights = post_op.depthwise.weights_data;
+            auto &biases = post_op.depthwise.biases_data;
+            postOpsDataPtrs.push_back(weights);
+            postOpsDataPtrs.push_back(biases);
+            weights = 0;
+            biases = 0;
+        }
+    }
+
+    auto buildExecutor = [&](const InterpolateKey& key) -> std::shared_ptr<InterpolateExecutor> {
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp:2149]: done, thanks Maksim.

File: src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp [hunk 4]
```diff
@@ -2038,21 +2130,73 @@ void MKLDNNInterpolateNode::prepareParams() {
...
+        }
+
+        auto* eltwiseNode = dynamic_cast<MKLDNNEltwiseNode *>(node.get());
+        if (eltwiseNode) {
+            eltwiseNode->appendPostOps(key.postOps, dstDims);
+            continue;
+        }
+
+        IE_THROW(Unexpected) << "Eltwise node with name '" << getName() << "' has unexpected fused op of type '" << node->getTypeStr() << "'";
+    }
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp]: done, thanks Maksim.

File: src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp [hunk 5]
```diff
@@ -1588,23 +1595,107 @@ struct jit_uni_interpolate_kernel_f32 : public jit_uni_interpolate_kernel, publi
...
+    seed = get_vector_hash(seed, nodeAttrs.padEnd);
+
+    seed = hash_combine(seed, nodeAttrs.inPrc.getPrecVal());
+    seed = hash_combine(seed, nodeAttrs.outPrc.getPrecVal());
+
+    seed = get_vector_hash(seed, srcDims);
+    seed = get_vector_hash(seed, dstDims);
+    seed = get_vector_hash(seed, dataScales);
+
+    // seed = hash_combine(seed, get_attr_hash(*attr.get()));
```
[the remark this answers is not in the document]
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp]: done, thanks Maksim.

File: src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp [hunk 6]
```diff
@@ -2038,20 +2125,51 @@ void MKLDNNInterpolateNode::prepareParams() {
...
+        std::shared_ptr<InterpolateExecutor> exector;
+        if ((key.nodeAttrs.mode == InterpolateMode::nearest || key.nodeAttrs.mode == InterpolateMode::linear_onnx ||
+            key.nodeAttrs.mode == InterpolateMode::cubic) &&
+            ((key.nodeAttrs.layout != InterpolateLayoutType::planar && mayiuse(cpu::x64::sse41)) ||
+                (mayiuse(cpu::x64::avx2) && key.nodeAttrs.inPrc == Precision::FP32))) {
+            exector = std::make_shared<InterpolateJitExecutor>(key.nodeAttrs,
+                                                               key.srcDims,
+                                                               key.dstDims,
+                                                               key.dataScales,
+                                                               attr);
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp:2159]: It seems more logical to use attr form the key, and the `attr` interpolate node data member can probably be removed as useless.

File: src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp [hunk 7]
```diff
@@ -2038,20 +2125,51 @@ void MKLDNNInterpolateNode::prepareParams() {
...
+            auto &biases = postOp.depthwise.biases_data;
+            postOpsDataPtrs.push_back(weights);
+            postOpsDataPtrs.push_back(biases);
+            weights = 0;
+            biases = 0;
+        }
+    }
+
+    auto buildExecutor = [&](const InterpolateKey& key) -> std::shared_ptr<InterpolateExecutor> {
+        std::shared_ptr<InterpolateExecutor> exector;
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp:2150]:
```suggestion
        std::shared_ptr<InterpolateExecutor> executor;
```

Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp]: fixed, thanks Maksim.

File: src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp (hunk 6 above)
Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/mkldnn_interpolate_node.cpp]: make sense, done, thanks Maksim.

**Review by maxnick — APPROVED** [review]

**Pull request merged**